### PR TITLE
update cugraph docker image for CI

### LINKED ISSUE #4914: Temporarily skip and re-enable CI for cugraphops
## 🔨Work Item

Due to some incompatibility issue between cugraph and cugraph-ops nightly, the cugraph docker image used in DGL CI is outdated and does not have the latest changes in cugraph-ops, and hence CI failed in PR #4826. Per our discussion in the core weekly meeting, we will temporarily disable the CI to get #4826 merged.

CC += @jermainewang

### PR DESCRIPTION
**Pull request by tingyu66**
## Description

Update cugraph docker image for CI. Fixes #4914.
Bump torch version to 2.0.0 and rapids' to 23.04.

Notes:
We start to use cugraph stable release from now on so that our breaking changes from nightly won't block any PRs from DGL. 

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

**Comment by tingyu66** [comment]
@VibhuJawa Can you take a look if I miss something?

File: Jenkinsfile [hunk 1]
```diff
@@ -545,7 +545,7 @@ pipeline {
               agent {
                 docker {
                   label "linux-gpu-node"
-                  image "rapidsai/cugraph_nightly_torch-cuda:11.5-base-ubuntu18.04-py3.9-pytorch1.12.0-rapids22.12"
+                  image "rapidsai/cugraph_nightly_torch-cuda:11.8-base-ubuntu20.04-py3.10-pytorch2.0.0-rapids23.04"
```
[the remark this answers is not in the document]
Comment (tingyu66) [reply Jenkinsfile:548]: Yep, let me change that

[the remark this answers is not in the document]
Comment (VibhuJawa) [reply Jenkinsfile]: Thanks. Resolved

**Comment by mufeili** [comment]
@dgl-bot

**Comment by mufeili** [comment]
@dgl-bot

**Comment by mufeili** [comment]
@tingyu66 Is this ready to be merged?

**Comment by tingyu66** [comment]
> @tingyu66 Is this ready to be merged?

Yes!

**Review by mufeili — APPROVED** [review]

**Pull request merged**